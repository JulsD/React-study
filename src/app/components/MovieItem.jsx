import React from 'react';

import btn from './../../styles/buttons.css';
import styles from './../../styles/movie-item.css';

class MovieItem extends React.Component {

  constructor(props) {
    super(props);

  }

  render() {
    const movieTitle = this.props.title;
    return (
      <div className = {styles.root}>
        <div className= {styles.cover}></div>
        <div className= {styles.details}>
          <div className= {styles.title}>{this.props.title} Title</div>
          <div className= {`${styles.date} ${btn.btn}`}>1990</div>
          <div className={styles.genre}>genre</div>
        </div>
      </div>
    );
  }

}

export default MovieItem;
