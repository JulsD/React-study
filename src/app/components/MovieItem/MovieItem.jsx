import React from 'react';
import { Link } from 'react-router-dom'

import btn from './../../../styles/buttons.css';
import styles from './movie-item.css';

class MovieItem extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Link to='/movie'>
        <div className = {styles.root}>
          <div className= {styles.cover}></div>
          <div className= {styles.details}>
            <div className= {styles.title}>{this.props.title} Title</div>
            <div className= {`${styles.date} ${btn.btn}`}>1990</div>
            <div className={styles.genre}>genre</div>
          </div>
        </div>
      </Link>
    );
  }

}

export default MovieItem;
