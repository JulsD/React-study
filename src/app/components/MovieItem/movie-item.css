.root {
  box-sizing: border-box;
  padding: 20px;
  min-width: 300px;
  display: inline-block;
  color:black;
}
.cover {
  position: relative;
  width: 100%;
  padding-top: 130%;
}
.cover div {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  z-index: -1;
}
.details {
  padding: 10px 5px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
}
.title {
  text-transform: uppercase;
  font-size: 15px;
  flex: 1 1 60%;
}
.genre {
  font-size: 11px;
  color: #666;
  flex: 1 1 100%;
}