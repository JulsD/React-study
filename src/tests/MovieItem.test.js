import React from 'react';
import MovieItem from '../app/components/MovieItem';

test('MovieItem should render a <Link>', () => {
  const movie = {
    poster_path: 'test',
    title: 'test',
    id: 'test',
    release_date: 'test',
    vote_average: 'test'
  }
  const wrapper = shallow(
    <MovieItem movie={movie}/>
  );
  expect(wrapper).toMatchSnapshot();
});
