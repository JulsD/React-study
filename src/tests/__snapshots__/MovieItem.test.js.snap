// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem should render a <Link> 1`] = `
<Link
  replace={false}
  to="/movie/test?title=test"
>
  <div
    className="root"
  >
    <div
      className="cover"
    >
      <div
        style={
          Object {
            "backgroundImage": "url(https://image.tmdb.org/t/p/w500test)",
          }
        }
      />
    </div>
    <div
      className="details"
    >
      <div
        className="title"
      >
        test
      </div>
      <Button
        kind="informative"
      >
        test
      </Button>
      <div
        className="genre"
      >
        test
         rate
      </div>
    </div>
  </div>
</Link>
`;
